Libraries/LayoutAnimation/LayoutAnimation.js

```diff
@@ -74,6 +74,12 @@ function configureNext(config: Config, onAnimationDidEnd?: Function) {
   UIManager.configureNextLayoutAnimation(
     config, onAnimationDidEnd || function() {}, function() { /* unused */ }
   );
+  // Clear the layout animation configuration after the current frame.
```

Comment:
I moved reseting the layout animation configuration to JS since it needs to be reset at the end of the JS frame and not only in the `_layoutAndMount` function. Not sure if there is a hook in native that gets called at the end of the JS frame it might be better there.